util/src/main/java/io/kubernetes/client/ProtoClient.java

```diff
@@ -69,43 +103,98 @@ public void setApiClient(ApiClient apiClient) {
      * @param path The URL path to call (e.g. /api/v1/namespaces/default/pods/pod-name)
      * @return A Message of type T
      */
-    public <T extends Message> T list(T.Builder listObj, String path) throws ApiException, IOException {
+    public <T extends Message> ObjectOrStatus<T> list(T.Builder listObj, String path) throws ApiException, IOException {
         return get(listObj, path);
     }
 
     /**
+     * Create a Kubernetes API object using protocol buffer encoding. Performs a POST
+     * @param obj The object to create
+     * @param path The URL path to call
+     * @param apiVersion The api version to use
+     * @param kind The kind of the object
+     * @return The response received.
+     */
+    public <T extends Message> ObjectOrStatus<T> create(T obj, String path, String apiVersion, String kind)
```

Comment:
Until we find a generic way to do it, I suggest we extract apiVersion and kind from T using reflection. I know it is not ideal, but I prefer it vs adding these two parameters to the surface of our proto APIs. ditto for other calls.

Reply:
Sadly, the fields aren't present in the proto.  See the long discussion here:

https://groups.google.com/forum/#!topic/kubernetes-sig-api-machinery/deP_LDaAdVs

Otherwise I would have used reflection.